src/helm/benchmark/scenarios/cleva_scenario.py

```diff
@@ -0,0 +1,1491 @@
+    Convert samples in CLEVA format to HELM instances according to CLEVA prompt template standard.
+    """
+
+    def transform(self, data: Dict[str, Any], templates: Dict[str, Any], split: str) -> Instance:
+        """Convert a data point in CLEVA format to a HELM instance according to a given CLEVA prompt template."""
+        transformed_data = self._apply_all(copy.deepcopy(data), templates)
+
+        text = transformed_data["input"]
+        if "choices" in transformed_data:
+            # This is a multiple-choice task
+            references: List[Reference] = [
+                Reference(Output(text=text), tags=[CORRECT_TAG] if idx in transformed_data["label"] else [])
+                for idx, text in enumerate(transformed_data["choices"])
+            ]
+        else:
+            # This is a generation task
+            correct_answer = transformed_data["label"]
+            references = [Reference(Output(text=answer), tags=[CORRECT_TAG]) for answer in correct_answer]
+
+        instance = Instance(
+            input=Input(text=text),
+            references=references,
+            split=split,
+        )
+        return instance
+
+    def transform_code(self, data: Dict[str, Any], templates: Dict[str, Any], split: str) -> CodeInstance:
+        """
+        Similar to transform method above, transform_code converts a data point in code synthesis scenario in CLEVA
+        to a HELM CodeInstance according to a given CLEVA prompt template.
+        """
+
+        data["prompt"] = templates["input"].format(**data)
+        instance = CodeInstance(
+            input=Input(text=data["prompt"]),
+            references=[
+                CodeReference(
+                    output=Output(text=data["canonical_solution"]),
+                    test_cases=data,
+                    tags=[CORRECT_TAG],
+                )
+            ],
+            split=split,
+        )
+        return instance
+
+    def _apply_all(self, data: dict, templates: dict) -> dict:
+        """
+        This function applies the CLEVA prompt template to a data point.
+
+        Note that this is an in-place operation.
+
+        The logic is as follows:
+        1. It first maps every entry according to a set of predefined mappings in "verbalizer".
+        2. It then stringifies all fields in the given data point, including processing structured data.
+        3. It finally constructs the input string and reference strings.
+        """
+        # If we define a verbalizer, we map all fields before further processing
+        if templates.get("verbalizer", None) is not None:
+            self._mapping_all(data, templates["verbalizer"])
+
+        if "choices" in data.keys():
+            # We take the corresonding choices and apply the `label` template
+            # Note: we do not allow `label` template to access other fields in multi-choice tasks
+            choices = [self._apply(c, templates.get("label", None), label=c) for c in data["choices"]]
```

Comment:
Can this be moved to right before `data["choices"] = choices`? Then you don't need two instances of `if "choices" in data.keys():`